tests/helper/helper.go

```diff
@@ -232,6 +232,29 @@ func WaitForJobCount(t *testing.T, kc *kubernetes.Clientset, namespace string,
 	return false
 }
 
+func WaitForJobCountUntilIteration(t *testing.T, kc *kubernetes.Clientset, namespace string,
+	target, iterations, intervalSeconds int) bool {
+	var isTargetAchieved = false
+
+	for i := 0; i < iterations; i++ {
+		jobList, _ := kc.BatchV1().Jobs(namespace).List(context.Background(), metav1.ListOptions{})
+		count := len(jobList.Items)
+
+		t.Logf("Waiting for job count to hit target. Namespace - %s, Current  - %d, Target - %d",
+			namespace, count, target)
+
+		if count == target {
+			isTargetAchieved = true
+		} else {
+			isTargetAchieved = false
+		}
+
+		time.Sleep(time.Duration(intervalSeconds) * time.Second)
+	}
+
+	return isTargetAchieved
+}
```

Comment:
I've added this function to ensure that no overscaling happens. The waitForJobCount function above in the file returns the result immediately as soon as` count == target`. This function waits for all iterations before returning the final result.

Reply:
/run-e2e external*